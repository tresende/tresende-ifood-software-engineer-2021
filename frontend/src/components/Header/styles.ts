import { makeStyles } from '@material-ui/core'

const useStyles = makeStyles((theme) => ({
  header: {
    margin: theme.spacing(1)
  },
  title: {
    fontFamily: "'Oleo Script', cursive",
    color: 'rgb(237, 23, 34)',
    userSelect: 'none'
  },
  italian: {
    textShadow: '5px -5px 0 rgb(0, 146, 70), 10px -10px 0 white, 15px -15px 0 rgb(206, 43, 55);'
  }
}))

export default useStyles
