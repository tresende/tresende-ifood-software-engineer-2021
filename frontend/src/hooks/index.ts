export * from './use-ifood-theme'
export * from './use-orders'
