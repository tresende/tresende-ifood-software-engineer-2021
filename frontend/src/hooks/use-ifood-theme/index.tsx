/* eslint-disable @typescript-eslint/no-empty-function */
import { useContext, createContext, useState } from 'react'
import { IFoodTheme, IFoodThemeContextData, IFoodThemeProviderProps } from './types'

const defaultValues: IFoodThemeContextData = {
  changeTheme: () => {},
  italianMode: false
}

export const IFoodThemeContext = createContext<IFoodThemeContextData>(defaultValues)

const IFoodThemeProvider = ({ children }: IFoodThemeProviderProps) => {
  const [theme, setTheme] = useState<IFoodTheme>('default')
  const changeTheme = () => setTheme(theme === 'default' ? 'italian' : 'default')

  const italianMode = theme === 'italian'
  return (
    <IFoodThemeContext.Provider
      value={{
        changeTheme,
        italianMode
      }}
    >
      {children}
    </IFoodThemeContext.Provider>
  )
}

const useIFoodTheme = () => useContext(IFoodThemeContext)

export { IFoodThemeProvider, useIFoodTheme }
