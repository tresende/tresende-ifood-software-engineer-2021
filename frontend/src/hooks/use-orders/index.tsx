/* eslint-disable @typescript-eslint/no-empty-function */
import { useContext, createContext, useState } from 'react'
import { OrdersContextData, OrdersProviderProps } from './types'
import { Filter, Order } from '../../types'
import { SearchOrders } from '../../services/OrderService'

const OrdersContextDefaultValues: OrdersContextData = {
  setFilter: () => {},
  search: () => {},
  orders: []
}

export const OrdersContext = createContext<OrdersContextData>(OrdersContextDefaultValues)

const OrdersProvider = ({ children }: OrdersProviderProps) => {
  const [orders, setOrders] = useState<Order[]>([])
  const [filter, setFilter] = useState<Filter>({})

  const search = async () => {
    try {
      const data = await SearchOrders(filter)
      setOrders(data)
    } catch {
      setOrders([])
    }
  }

  return (
    <OrdersContext.Provider
      value={{
        orders,
        setFilter,
        search
      }}
    >
      {children}
    </OrdersContext.Provider>
  )
}

const useOrders = () => useContext(OrdersContext)

export { OrdersProvider, useOrders }
