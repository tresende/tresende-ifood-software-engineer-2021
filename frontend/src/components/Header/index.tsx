import { Typography } from '@material-ui/core'
import { useIFoodTheme } from '../../hooks'
import useStyles from './styles'

const Header = () => {
  const { italianMode } = useIFoodTheme()
  const classes = useStyles()
  const titleClasses = [classes.title]
  if (italianMode) {
    titleClasses.push(classes.italian)
  }
  return (
    <header className={classes.header}>
      <Typography className={titleClasses.join(' ')} variant="h2" component="h2">
        Order List
      </Typography>
    </header>
  )
}

export default Header
