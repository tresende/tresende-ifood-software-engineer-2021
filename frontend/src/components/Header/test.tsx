import { render } from '@testing-library/react'
import { useIFoodTheme } from 'hooks'

import Header from '.'

jest.mock('../../hooks')

describe('<Header />', () => {
  it('should take snapshot with italian mode', () => {
    ;(useIFoodTheme as jest.Mock).mockReturnValue({ italianMode: true })

    const { container } = render(<Header />)

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should take snapshot with default mode', () => {
    ;(useIFoodTheme as jest.Mock).mockReturnValue({ italianMode: false })

    const { container } = render(<Header />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
