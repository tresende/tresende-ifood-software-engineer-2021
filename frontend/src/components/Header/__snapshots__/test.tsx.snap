// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should take snapshot with default mode 1`] = `
<header
  class="makeStyles-header-4"
>
  <h2
    class="MuiTypography-root makeStyles-title-5 MuiTypography-h2"
  >
    Order List
  </h2>
</header>
`;

exports[`<Header /> should take snapshot with italian mode 1`] = `
<header
  class="makeStyles-header-1"
>
  <h2
    class="MuiTypography-root makeStyles-title-2 makeStyles-italian-3 MuiTypography-h2"
  >
    Order List
  </h2>
</header>
`;
